test/testRNode.js

```diff
@@ -1,139 +1,144 @@
-const Suite = require('testjs');
-const grpc = require('grpc');
-
-const { RNode, RHOCore, b2h } = require('../rnodeAPI');
+/* global require, module, exports */
+const rnode = require('../rnodeAPI');
+
+const { RNode, RHOCore, b2h } = rnode;
+const { sha256Hash, keccak256Hash, blake2b256Hash } = rnode;
+const { simplifiedSHA256Hash, simplifiedKeccak256Hash, simplifiedBlake2b256Hash } = rnode;
+
+
+/**
+ * Run unit tests plus supplemental tests.
+ *
+ * Suite.run() can only be called in one place because
+ * it does `process.exit()`. Using ocap discipline
+ * means passing such powerful objects in explicitly.
+ *
+ * @param suite2: supplemental tests
+ */
+function testRNode({ Suite }, suite2) {
+  // mock enough of grpc
+  function DeployService(_hostPort, _chan) { }
+  const casper = { DeployService };
+  const proto = { coop: { rchain: { casper: { protocol: casper } } } };
+  const grpc0 = {
+    loadPackageDefinition(_d) { return proto; },
+    credentials: { createInsecure() { } },
+  };
 
-function testRNode() {
   Suite.run({
     'args check': (test) => {
-      test.doesNotThrow(() => RNode(grpc, { host: 'h', port: 123 }));
-      test.throws(() => RNode(null, null), Error);
-      test.throws(() => RNode(null, { host: 'hi' }), Error);
-      test.throws(() => RNode(null, { port: 123 }), Error);
+      test.doesNotThrow(() => RNode(grpc0, { host: 'h', port: 123 }));
+      // $FlowFixMe args are intentionally wrong type
+      test.throws(() => RNode(grpc0, null), Error);
+      // $FlowFixMe
+      test.throws(() => RNode(grpc0, { host: 'hi' }), Error);
+      // $FlowFixMe
+      test.throws(() => RNode(grpc0, { port: 123 }), Error);
       test.done();
     },
+    ...suite2,
+  });
+}
+
+
+function netTests({ grpc, clock, rng }) {
+  const localNode = () => RNode(grpc, { host: 'localhost', port: 40401 });
+
+  function simplifiedHashTest(test, fn, fname) {
+    const returnChannel = rng().toString(36).substring(7);
+    const txt = 'test';
+    const hashProc = `@"${fname}"!("${txt}".toByteArray(), "${returnChannel}")`;
+
+    runAndListen(hashProc, returnChannel, clock().valueOf(), localNode(), test)
+      .then((rholangHash) => {
+        test.equal(fn('test'), b2h(rholangHash.exprs[0].g_byte_array));
+        test.done();
+      })
+      .catch((oops) => {
+        test.equal(oops, 0);
+        test.done();
+      });
+  }
+
+  function normalHashTest(test, fn, fname) {
```

Comment:
There's a lot of repeated code between `simplifiedHashTest` and `normalHashTest`. Not critical in test code, but not good for maintenance.

Reply:
Refactored / consolidated these functions.  Differentiating the tests by adding an "isNormalTest" argument to the hashTest function.